fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/ChirpHandler.java

```diff
@@ -28,14 +29,19 @@ private ChirpHandler() {
    * @param context the HandlerContext of the message
    * @param addChirp the data of the message that was received
    */
-  public static void handleChirpAdd(HandlerContext context, AddChirp addChirp) {
+  public static void handleChirpAdd(HandlerContext context, AddChirp addChirp)
+      throws UnknownLaoException {
     LAORepository laoRepository = context.getLaoRepository();
     Channel channel = context.getChannel();
     MessageID messageId = context.getMessageId();
     PublicKey senderPk = context.getSenderPk();
 
-    Lao lao = laoRepository.getLaoByChannel(channel);
-
+    Log.d(TAG, "handleChirpAdd: " + channel + " id " + addChirp.getParentId());
+    Optional<LaoView> laoViewOptional = laoRepository.getLaoViewByChannel(channel);
+    if (!laoViewOptional.isPresent()) {
+      throw new UnknownLaoException(channel.extractLaoId());
+    }
+    LaoView laoView = laoViewOptional.get();
```

Comment:
Would it make sense to create a function inside the repository that throws an exception when no Lao are found ?

It seems like a usual usage of the repository.
Btw, you can write it like this :
```suggestion
    LaoView laoView = laoViewOptional.orElseThrow(() -> new UnknownLaoException(channel.extractLaoId()));
```

Reply:
For now it is the only behaviour we have when getting an LAO from the repo so it seems like a reasonable suggestion to me.